programs/drift_vaults/src/state/vault_depositor.rs

```diff
@@ -69,6 +69,104 @@ const_assert_eq!(
     std::mem::size_of::<VaultDepositor>() + 8
 );
 
+impl VaultDepositorBase for VaultDepositor {
+    fn apply_rebase(
+        self: &mut VaultDepositor,
+        vault: &mut Vault,
+        vault_protocol: &mut Option<RefMut<VaultProtocol>>,
+        vault_equity: u64,
+    ) -> Result<Option<u128>> {
+        vault.apply_rebase(vault_protocol, vault_equity)?;
+
+        let mut rebase_divisor = None;
+
+        if vault.shares_base != self.vault_shares_base {
+            validate!(
+                vault.shares_base > self.vault_shares_base,
+                ErrorCode::InvalidVaultRebase,
+                "Rebase expo out of bounds"
+            )?;
+
+            let expo_diff = (vault.shares_base - self.vault_shares_base).cast::<u32>()?;
+
+            let _rebase_divisor = 10_u128.pow(expo_diff);
+
+            msg!(
+                "rebasing vault depositor: base: {} -> {} ",
+                self.vault_shares_base,
+                vault.shares_base,
+            );
+
+            self.vault_shares_base = vault.shares_base;
+
+            let old_vault_shares = self.unchecked_vault_shares();
+            let new_vault_shares = old_vault_shares.safe_div(_rebase_divisor)?;
+
+            msg!(
+                "rebasing vault depositor: shares {} -> {} ",
+                old_vault_shares,
+                new_vault_shares
+            );
+
+            self.update_vault_shares(new_vault_shares, vault)?;
+
+            self.last_withdraw_request.rebase(_rebase_divisor)?;
```

Comment:
the main difference between apply rebase for tokenized and non tokenized vault is that normal deposit rebases last withdraw request and tokenized depositor updates last_vault_shares?

is there any way to mv duplicate code into one place so its clearer code is duplicate?